### PR TITLE
Guy/webpack optimization

### PR DESCRIPTION
**Pull request by guyyoo**
https://github.com/digi-serve/test_pwa/issues/4

Other work:
Nginx detect mobile device and redirect to PWA app.

### CODE REVIEW

File: default.conf [hunk 1]
```diff
@@ -34,12 +45,30 @@ server {
     # /home
     # redirect to /app/assets/index.html
     location = /home {
-        add_header Cache-Control "no-cache";
+        if ($is_mobile = "1") {
+            return 301 $scheme://$http_host/mobilehome;
```
Comment (nh758) [inline default.conf:49]: I'm not sure if we want this to redirect to the pwa?

**Review by nh758 — COMMENTED** [review]
Mostly looks good. I'm just not sure if the nginx config is the way we want to do it. Wait for @Hiro-Nakamura to review 😄

File: default.conf (hunk 1 above)
Comment (Hiro-Nakamura) [reply default.conf:49]: to be honest, i'm not sure what `/mobilehome` is.  Is this a new route?  

does it lead to a Mobile friendly version of our Web app?  Or is it supposed to link to a specific PWA?  If so, how does it know which one?

File: default.conf [hunk 2]
```diff
@@ -34,12 +45,30 @@ server {
...
+    location = /mobilehome {
+        if ($is_mobile = "0") {
+            return 301 $scheme://$http_host/home;
+        }
+
+        # don't cache so the most uptodate js files are requested
+        add_header Cache-Control "no-cache";
+
+        root /app/assets;
+        try_files /mobile/index.html =404;
```
Comment (Hiro-Nakamura) [inline default.conf:69]: yeah, so for a PWA app ... they should be served up by the `api_sails/views/mobile_pwa.ejs` file.  Not this `/mobile/index.html`.

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
I've noticed that the web pack generates a bunch of these `mobile_src_js_AppBuilder_... .js` files  that also get loaded with the app.  Can we simply keep everything contained in the 3 main `.js` files and get rid of these smaller files?

File: default.conf (hunk 1 above)
Comment (guyyoo) [reply default.conf:49]: My mistake was thinking that PWA = mobile. This should actually link to the PWA. Can you suggest what this should be?

File: default.conf (hunk 2 above)
Comment (guyyoo) [reply default.conf:69]: Okay, thanks!

File: default.conf (hunk 1 above)
Comment (guyyoo) [reply default.conf]: https://github.com/digi-serve/ab_service_web/pull/11/commits/079d1fd3f62b4c04a87fd61c7b33c6fc9e4c668b

File: assets/mobile/icons/256x256.png
[the remark this answers is not in the document]
Comment (guyyoo) [reply assets/mobile/icons/256x256.png:1]: https://github.com/digi-serve/ab_service_web/pull/11/commits/9fe1f7f7a71765bdd855b2fc49d7735768379846
https://github.com/digi-serve/ab_service_web/pull/11/commits/31351b09e52a41bfb539628687ddd0bd8541bf5b

**Review by Hiro-Nakamura — APPROVED** [review]
LGTM

**Pull request merged**